codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/UnconstrainedShapeSymbolProvider.kt

```diff
@@ -75,22 +79,42 @@ import software.amazon.smithy.rust.codegen.server.smithy.generators.serverBuilde
 class UnconstrainedShapeSymbolProvider(
     private val base: RustSymbolProvider,
     private val model: Model,
-    private val serviceShape: ServiceShape,
     private val publicConstrainedTypes: Boolean,
+    private val serviceShape: ServiceShape,
 ) : WrappingSymbolProvider(base) {
     private val nullableIndex = NullableIndex.of(model)
+    private val visibility = when (publicConstrainedTypes) {
+        true -> Visibility.PUBLIC
+        false -> Visibility.PUBCRATE
+    }
+
+    /**
+     * Unconstrained type names are always suffixed with `Unconstrained` for clarity, even though we could dispense with it
+     * given that they all live inside the `unconstrained` module, so they don't collide with the constrained types.
+     */
+    private fun unconstrainedTypeNameForCollectionOrMapOrUnionShape(shape: Shape): String {
+        check(shape is CollectionShape || shape is MapShape || shape is UnionShape)
+        return RustReservedWords.escapeIfNeeded(shape.contextName(serviceShape).toPascalCase() + "Unconstrained")
+    }
 
+    // this is required because the `Name` field on the symbol is `Vec` (since usually it would be rendered as `Vec<Foo>`
```

Comment:
I don't understand this comment. The name field on the symbol should coincide with the Rust type name, which is the shape name suffixed with `Unconstrained`.

Reply:
I will clarify—this is to explain why you can't use the base symbol (which you could do for a structure shape).

It's because in this case:
```
list FooList {
  member: Bar
}
```

The resulting symbol is `Vec<Bar>` _not_ `FooList` (but if you need something named `FooList` you have to take a different path)

Reply:
Yep, that's right. We need to generate unconstrained structs that are newtypes (of e.g. `Vec<Bar>` in your example) for anything that transitively reaches a constrained shape.